### DISCUSSION TITLE
v7.3.0-preview.5 Release of PowerShell

### DISCUSSION THREAD

**Discussion by adityapatwardhan** [PowerShell/PowerShell#17562] [General]
### Engine Updates and Fixes

- Improve type inference and completions (#16963) (Thanks @MartinGC94!)
- Make `Out-String` and `Out-File` keep string input unchanged (#17455)
- Make `AnsiRegex` able to capture Hyperlink ANSI sequences (#17442)
- Add the `-ConfigurationFile` command line parameter to `pwsh` to support local session configuration (#17447)
- Fix native library loading for `osx-arm64` (#17365) (Thanks @awakecoding!)
- Fix formatting to act appropriately when the style of table header or list label is empty string (#17463)

### General Cmdlet Updates and Fixes

- Fix various completion issues inside the `param` block (#17489) (Thanks @MartinGC94!)
- Add Amended switch to `Get-CimClass` cmdlet (#17477) (Thanks @iSazonov!)
- Improve completion on operators (#17486) (Thanks @MartinGC94!)
- Improve array element completion for command arguments (#17078) (Thanks @matt9ucci!)
- Use AST extent for `PSScriptRoot` path completion (#17376)
- Add type inference support for generic methods with type parameters (#16951) (Thanks @MartinGC94!)
- Write out OSC indicator only if the `stdout` is not redirected (#17419)
- Remove the assert and use a relatively larger capacity to cover possible increase of .NET reference assemblies (#17423)
- Increase reference assembly count to 161 (#17420)

### Code Cleanup

<details>

<summary>

<p>We thank the following contributors!</p>
<p>@Yulv-git, @eltociear</p>

</summary>

<ul>
<li>Fix some typos in source code (#17481) (Thanks @Yulv-git!)</li>
<li>Fix typo in `AsyncResult.cs` (#17396) (Thanks @eltociear!)</li>
</ul>

</details>

### Tools

- Update script to pin to .NET 7 preview 5 version (#17448)
- Start-PSPester: argument completer for `-Path` (#17334) (Thanks @powercode!)
- Add reminder workflows (#17387)
- Move to configuring the fabric bot via JSON (#17411)
- Update Documentation Issue Template URL (#17410) (Thanks @michaeltlombardi!)
- Update script to automatically take new preview pre-release builds (#17375)

### Tests

- Make Assembly Load Native test work on a FX Dependent Linux Install (#17380)
- Update `Get-Error` test to not depend on DNS APIs (#17471)

### Build and Packaging Improvements

<details>

<ul>
<li>Update .NET SDK version from <code>7.0.100-preview.4.22252.9</code> to <code>7.0.100-preview.5.22307.18</code> (#17402)</li>
<li>Downgrade the <code>Microsoft.CodeAnalysis.NetAnalyzers</code> package to <code>7.0.0-preview1.22217.1</code> (#17515)</li>
<li>Rename mariner package to cm (#17505)</li>
<li>Bump Microsoft.CodeAnalysis.NetAnalyzers (#17476)</li>
<li>Bump <code>NJsonSchema</code> from 10.7.1 to 10.7.2 (#17475)</li>
<li>Publish preview versions of mariner to preview repo (#17451)</li>
<li>Update to the latest NOTICES file (#17421)</li>
<li>Do not publish package for Mariner 1.0 (#17415)</li>
<li>Add <code>AppX</code> capabilities in MSIX manifest so that PS7 can call the <code>AppX</code> APIs (#17416)</li>
<li>Update to the latest NOTICES file (#17401)</li>
<li>Fix mariner mappings (#17413)</li>
<li>Update the cgmanifest (#17393)</li>
<li>Bump `NJsonSchema` from `10.7.0` to `10.7.1` (#17381)</li>
</ul>

</details>

### Documentation and Help Content

- Update to the latest NOTICES file (#17493) (Thanks @github-actions[bot]!)
- Update the cgmanifest (#17478) (Thanks @github-actions[bot]!)
- Correct spelling in Comments and tests (#17480) (Thanks @Yulv-git!)
- Fix spelling errors introduced in changelog (#17414)
- Update change log for v7.3.0-preview.4 release (#17412)
- Update readme and metadata for 7.3.0-preview.4 release (#17378)

[7.3.0-preview.5]: https://github.com/PowerShell/PowerShell/compare/v7.3.0-preview.4...v7.3.0-preview.5

### SHA256 Hashes of the release artifacts

- hashes.sha256
  - B60B5DC6C66F8FC737EC856800E827AB4DBC418092905D8459A4EE122A667264
- powershell-7.3.0-preview.5-linux-alpine-x64.tar.gz
  - 692896BFF922EA3C707B67A6EFA70EDFB1336D78B939121C3FDB41E12DB80449
- powershell-7.3.0-preview.5-linux-arm32.tar.gz
  - F40D9F1616CF62870A974E7A1089A2E82220C50CD1F6468DD0B2FB1ED3E509CE
- powershell-7.3.0-preview.5-linux-arm64.tar.gz
  - 5ADB98F5AAD1B58DB95794448071628DCA7AEF861797C6AB4B9B861A47B9FB81
- powershell-7.3.0-preview.5-linux-x64.tar.gz
  - 67F68092CB5BA914777D7456A0ADE153DFFC347DBE014C9086DB3DD87B00B847
- powershell-7.3.0-preview.5-linux-x64-fxdependent.tar.gz
  - 09016B402C3530B20F20F8B004C80ABB46863B3B503BE097DA8946231AA4ABA3
- powershell-7.3.0-preview.5-osx-arm64.pkg
  - A7C28FF9CE49A5DA842208B8A9B559CCEBD04E78A32794ED9DEAEF98DC27BBB0
- powershell-7.3.0-preview.5-osx-arm64.tar.gz
  - E59EB0BF686388F9FE2681F2A3D4C406A8A8162CD8989A448B3AC802E9882D84
- powershell-7.3.0-preview.5-osx-x64.pkg
  - 7A6C7BD60D6F15C45D83E9045E52FEFBB21877BA9EDA814FF7F817B4229A7096
- powershell-7.3.0-preview.5-osx-x64.tar.gz
  - EF79B4C8A115E22B8F5ED2CD69EA2DCACFA93703A2BAE04A48C8E898EA1C8941
- PowerShell-7.3.0-preview.5-win-arm32.zip
  - 12706E4F1ED3CF1210D2305B9AF8AD6794A5CBD736798E30E7F20E559AB8CB3E
- PowerShell-7.3.0-preview.5-win-arm64.zip
  - 809CAB5EFCBEEE21C219167255AE12810330ADDB0FDCB6ABA751619F4E377491
- PowerShell-7.3.0-preview.5-win-fxdependent.zip
  - 4A349E2F3ACF12A5E959DEB72F32AE1303DD1D712C92D0873D58F2278EC54BFB
- PowerShell-7.3.0-preview.5-win-fxdependentWinDesktop.zip
  - 390BBE2CCB20DABD3EF9E7710F9FAB1D254B4D64B3BACFBFC41516DEF40231A4
- PowerShell-7.3.0-preview.5-win-x64.msi
  - D1F297068A4830D8FDF623A1F038A423DE94804EF974F1BAF88DEAC4D24CE628
- PowerShell-7.3.0-preview.5-win-x64.zip
  - B1E44525E211EC35FE0772314B0C2F732768418B173AE5E9BAD2300342D03960
- PowerShell-7.3.0-preview.5-win-x86.msi
  - 2B23333D56E95EE5508818DD0EAEBBF61260312A2DC0DCA2F578C7ED217D5B4B
- PowerShell-7.3.0-preview.5-win-x86.zip
  - DBABF906F36C7BEF920733CCDD747BCF36499C41A4CAC4066DF3709055CAE299
- powershell-preview_7.3.0-preview.5-1.deb_amd64.deb
  - C6294B5E3F4CCD0BE66FBBAA85C712A6E8EA40B6B1EF21B2C0FA75665B2A2BBD
- powershell-preview-7.3.0_preview.5-1.cm.x86_64.rpm
  - 50B9794810977A39DEEF383B9F00FF64F0A2ABE2422C00BB4CF783BC377B840D
- powershell-preview-7.3.0_preview.5-1.rh.x86_64.rpm
  - EE75D1B7C9540501890063727733CB5A3DA59C977B6EDC6F5F173D656A7ED407

<hr /><em>This discussion was created from the release <a href='https://github.com/PowerShell/PowerShell/releases/tag/v7.3.0-preview.5'>v7.3.0-preview.5 Release of PowerShell</a>.</em>

**Comment by hongwen000** [comment]
Is this version already built on dotnet 7?

**Reply by rkeithhill** [reply]
Yes. 
```powershell
PS> [System.Runtime.InteropServices.RuntimeInformation]::FrameworkDescription
.NET 7.0.0-preview.5.22301.12
```

**Comment by hongwen000** [comment]
That’s awesome!

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows
